src/components/settings.jsx

```diff
@@ -55,26 +54,38 @@ class Settings extends Component {
             else {
                 console.log('ERROR')
             }
+            this.setState({change: true})
     }
     handleChangeUserName = (e) => {
-        this.setState({ username: e.target.value })
-
+        this.setState({ 
+            username: e.target.value,
+            change: true
+        })
     }
     handleFirstName = (e) => {
-        this.setState({ fName: e.target.value })
+        this.setState({ 
+            fName: e.target.value,
+            change: true
+        })
     }
     handleLastName = (e) => {
-        this.setState({ lName: e.target.value })
+        this.setState({ 
+            lName: e.target.value,
+            change: true
+        })
     }
     handleEMAIL = (e) => {
-        this.setState({ email: e.target.value })
+        this.setState({ 
+            email: e.target.value,
+            change: true
+        })
     }
-
     handleBio = (e) => {
-        this.setState({ bio: e.target.value })
+        this.setState({ 
+            bio: e.target.value,
+            change: true 
+        })
     }
-    
-
     handleSubmit = (e) => {
```

Comment:
lets put the firebase update in the "if (this.state.change)" too. since we dont want to update if nothing changed